domehunter/gRPC-server/hx2dome.proto_server.py

```diff
@@ -0,0 +1,578 @@
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            return_code = 0
+            try:
+                self.dome.abort()
+            except:
+                # TODO: proper error handling
+                return_code = 1
+            response = hx2dome_pb2.ReturnCode(return_code=return_code)
+            return response
+
+    def dapiOpen(self, request, context):
+        """TheSkyX RPC to open dome slit.
+        NOTE: this functionality isn't currently implemented.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+
+    def dapiClose(self, request, context):
+        """TheSkyX RPC to close dome slit.
+        NOTE: this functionality isn't currently implemented.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+
+    def dapiPark(self, request, context):
+        """TheSkyX RPC to park the dome.
+        NOTE: this functionality isn't currently implemented.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            # TODO: unsure what action to trigger in automationHAT for "Park"
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+
+    def dapiUnpark(self, request, context):
+        """TheSkyX RPC to unpark the dome.
+        NOTE: this functionality isn't currently implemented.
```

Comment:
@fergusL I think you commented on this somewhere, but yes it seems like the `Dome` should have a property that tracks if it is parked or not and ignores movement commands if so.

Reply:
ah ok, yea I opened an issue (#28) because I was unsure but that does seem like the right idea.